### PR TITLE
Publish nightly builds

### PR DESCRIPTION
**Pull request by mischnic**
Closes https://github.com/adobe/react-spectrum/issues/723

Done:

- Would run daily at 09:00 UTC = 02:00 PDT
- Upgrade Parcel version
- Silenced a few typescript errors that came up since my last PR

Questions:

- The current .npmrc files (`.npmrc` and `.circle/.npmrc`) contain some internal registries, is that still needed? Not sure how it would behave at the moment

- With Parcel, we have the problem that, according to semver, and `2.0.0-nightly.352` matches `parcel@^2.0.0-beta.1`...
- Should all packages be published, or only those with changes?
- ~Might want to hold off until the double-build situation is fixed in Parcel (https://github.com/parcel-
bundler/parcel/pull/4887)~

## 📝 Test Instructions:

Can be tested by running a Verdaccio registry locally with this config (`verdaccio -c file.yml`)
```yml
storage: /Users/niklas/.local/share/verdaccio/storage
plugins: ./plugins

web:
  title: Verdaccio

auth:
  htpasswd:
    file: ./htpasswd

uplinks:
  npmjs:
    url: https://registry.npmjs.org/

packages:
  '@adobe/react-spectrum':
    access: $all
    publish: $all
    unpublish: $all
  
  '@react-spectrum/*':
    access: $all
    publish: $all
    unpublish: $all
  
  '@react-aria/*':
    access: $all
    publish: $all
    unpublish: $all
  
  '@react-stately/*':
    access: $all
    publish: $all
    unpublish: $all
  
  '@react-types/*':
    access: $all
    publish: $all
    unpublish: $all
  
  '@spectrum-icons/*':
    access: $all
    publish: $all
    unpublish: $all

  '@*/*':
    access: $all
    publish: $authenticated
    unpublish: $authenticated
    proxy: npmjs

  '**':
    access: $all
    publish: $authenticated
    unpublish: $authenticated
    proxy: npmjs

server:
  keepAliveTimeout: 60

logs:
  - { type: stdout, format: pretty, level: http }
```
and
```diff
diff --git package.json package.json
index ebd9898b..8402b9fd 100644
--- package.json
+++ package.json
@@ -29,7 +29,7 @@
     "start:docs": "PARCEL_WORKER_BACKEND=process parcel 'packages/@react-{spectrum,aria,stately}/*/docs/*.mdx' 'packages/dev/docs/pages/**/*.mdx'",
     "build:docs": "PARCEL_WORKER_BACKEND=process parcel build 'packages/@react-{spectrum,aria,stately}/*/docs/*.mdx' 'packages/dev/docs/pages/**/*.mdx' --no-scope-hoist",
     "publish": "lerna publish from-package --yes",
-    "publish:nightly": "lerna publish -y --canary --preid nightly --dist-tag=nightly --exact --force-publish=* --no-push"
+    "publish:nightly": "lerna publish -y --canary --preid nightly --dist-tag=nightly --exact --force-publish=* --no-push --registry http://localhost:4873/"
   },
   "workspaces": [
     "packages/*/*"
```

After running `make publish-nightly`, create a new project and installing/using the packages after running `npm set registry http://localhost:4873/`.
Verdaccio also serves a browser UI at http://localhost:4873/

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -157,29 +157,43 @@ jobs:
...
+          name: Authenticate with npm
+          command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/react-spectrum/.npmrc
+      - run:
+          name: Publish
+          command: make publish-nightly
+
 workflows:
   version: 2
-  test:
+  commit:
```
Comment (snowystinger) [inline .circleci/config.yml:176]: 👍

**Comment by bhumip214** [comment]
Would this change also publish private packages in nightlies?

**Comment by mischnic** [comment]
> Would this change also publish private packages in nightlies?

No

**Comment by devongovett** [comment]
Tried to upgrade parcel to the latest nightly again and failed miserably. 😆 

It builds but I get "Uncaught Error: Cannot find module '1t8mc'" at runtime. Seems to be related to react fast refresh. Didn't have time to look into it more...

**Pull request merged**